### PR TITLE
fix: make all exit calls async

### PR DESCRIPTION
**Pull request by colinnielsen**
## Checklist

- [x] Documentation updated
- [x] Tested code changes

[ticket](https://linear.app/metropolis/issue/MP-547/make-exit-calls-async)

### CODE REVIEW

File: src/commands/deploy.ts [hunk 1]
```diff
@@ -49,56 +38,10 @@ export const builder: { [key: string]: Options } = {
...
-    etherscanUrl: undefined,
-  };
-
-  return doNotCommunicateWithMetalService
-    ? emptyConfig
-    : !!userHasSpecifiedRPC
-    ? { ...emptyConfig, rpcUrl: getFlagValueFromArgv(process.argv[rpcOverrideFlagIdx + 1]) }
-    : await fetchChainConfig(chainId);
-};
-
```
Comment (kevinweaver) [inline src/commands/deploy.ts:97]: Good call moving these to a helper

**Review by kevinweaver — APPROVED** [review]
👍 I didn't get a chance to run this locally. If you think it's worth it for the sanity check, I'm happy to tomorrow.

File: src/commands/deploy.ts (hunk 1 above)
Comment (colinnielsen) [reply src/commands/deploy.ts:97]: ty ty

**Comment by colinnielsen** [comment]
> 👍 I didn't get a chance to run this locally. If you think it's worth it for the sanity check, I'm happy to tomorrow.

Tested

**Pull request merged**